tensorflow/compiler/tf2tensorrt/convert/op_converter_registry.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/compiler/tf2tensorrt/convert/op_converter_registry.h"
+
+#include <set>
+#include <utility>
+
+#include "tensorflow/core/platform/mutex.h"
+
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+
+namespace tensorflow {
+namespace tensorrt {
+namespace convert {
+
+struct OpConverterRegistration {
+  OpConverter converter;
+  int priority;
+};
+class OpConverterRegistry::Impl {
+ public:
+  ~Impl() = default;
+
+  InitOnStartupMarker Register(const string& name, const int priority,
+                               OpConverter converter) {
+    mutex_lock lock(mu_);
+    auto set = registry_.find(name);
+    if (set == registry_.end()) {
+      registry_.insert({name, OpConverterRegistration{converter, priority}});
+    } else {
+      if (set->second.priority < priority) {
+        LOG(WARNING) << absl::StrCat(
+            "Overriding TF->TRT ",
+            set->second.priority == kDefaultConverterPriority
+                ? "default op converter"
+                : " non-default op converter",
+            " for op ", name, " with converter of priority ", priority);
+      }
```

Comment:
Sorry, I don't see the step that you overwrite the value in registry_ with the new converter?
Also, if the new converter is not used to overwrite the content in registry_, we also need to issue a warning.

Reply:
yes, I'm going to push a commit here which adds a test and fixes this

Reply:
Fixed